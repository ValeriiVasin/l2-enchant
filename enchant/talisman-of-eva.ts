import { createEnchantMap } from '../helpers/create-enchant-map';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const talismanOfEvaEnchantMap = createEnchantMap({
  strategy: EnchantStrategy.Destroy,
  chances: [60, 50, 40, 35, 30, 28, 26, 24, 22, 20],
  items: [
    Item.TalismanOfEva,
    Item.TalismanOfEva_1,
    Item.TalismanOfEva_2,
    Item.TalismanOfEva_3,
    Item.TalismanOfEva_4,
    Item.TalismanOfEva_5,
    Item.TalismanOfEva_6,
    Item.TalismanOfEva_7,
    Item.TalismanOfEva_8,
    Item.TalismanOfEva_9,
    Item.TalismanOfEva_10,
  ],
});
