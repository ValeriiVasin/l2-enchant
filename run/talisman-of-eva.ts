import { run } from '../helpers/run';
import { Item } from '../items';

run(
  Item.TalismanOfEva,
  [Item.TalismanOfEva_3, Item.TalismanOfEva_4, Item.TalismanOfEva_5, Item.TalismanOfEva_6, Item.TalismanOfEva_7],
  10_000,
);
