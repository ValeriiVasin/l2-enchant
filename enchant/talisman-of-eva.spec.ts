import { talismanOfEvaEnchantMap } from './talisman-of-eva';

it('enchant map: talisman of eva', () => {
  expect(talismanOfEvaEnchantMap).toMatchSnapshot();
});
