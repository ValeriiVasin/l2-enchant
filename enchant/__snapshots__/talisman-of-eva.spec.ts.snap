// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: talisman of eva 1`] = `
Map {
  "Талисман Евы" => Object {
    "cost": "Nothing",
    "item": "Талисман Евы",
    "results": Array [
      Object {
        "chance": 60,
        "item": "Талисман Евы +1",
      },
      Object {
        "chance": 40,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Евы +1" => Object {
    "cost": "Nothing",
    "item": "Талисман Евы +1",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Евы +2",
      },
      Object {
        "chance": 50,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Евы +2" => Object {
    "cost": "Nothing",
    "item": "Талисман Евы +2",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Талисман Евы +3",
      },
      Object {
        "chance": 60,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Евы +3" => Object {
    "cost": "Nothing",
    "item": "Талисман Евы +3",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Талисман Евы +4",
      },
      Object {
        "chance": 65,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Евы +4" => Object {
    "cost": "Nothing",
    "item": "Талисман Евы +4",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Талисман Евы +5",
      },
      Object {
        "chance": 70,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Евы +5" => Object {
    "cost": "Nothing",
    "item": "Талисман Евы +5",
    "results": Array [
      Object {
        "chance": 28,
        "item": "Талисман Евы +6",
      },
      Object {
        "chance": 72,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Евы +6" => Object {
    "cost": "Nothing",
    "item": "Талисман Евы +6",
    "results": Array [
      Object {
        "chance": 26,
        "item": "Талисман Евы +7",
      },
      Object {
        "chance": 74,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Евы +7" => Object {
    "cost": "Nothing",
    "item": "Талисман Евы +7",
    "results": Array [
      Object {
        "chance": 24,
        "item": "Талисман Евы +8",
      },
      Object {
        "chance": 76,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Евы +8" => Object {
    "cost": "Nothing",
    "item": "Талисман Евы +8",
    "results": Array [
      Object {
        "chance": 22,
        "item": "Талисман Евы +9",
      },
      Object {
        "chance": 78,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Евы +9" => Object {
    "cost": "Nothing",
    "item": "Талисман Евы +9",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Талисман Евы +10",
      },
      Object {
        "chance": 80,
        "item": "Nothing",
      },
    ],
  },
}
`;
